### DISCUSSION TITLE
Moving (some of) `src` to TypeScript

### DISCUSSION THREAD

**Discussion by schickling** [rhashimoto/wa-sqlite#205] [General]
Hi @rhashimoto,

As part of https://github.com/rhashimoto/wa-sqlite/discussions/200, I've been recently spending more time in the wa-sqlite codebase. I really like the clear structure and generally high quality of code - well done!

Given most of the projects I'm working on are developed and maintained in TypeScript and then through a build-step compiled to JS + type definitions, I was wondering whether migrating to this approach (which I'd consider "industry standard") is something you'd be open to? I'd be happy to help with this effort and it's also something that could be done incrementally.

If you'd prefer to keep the setup like it is right now, no problem, just figured I'd ask and offer my help with this :)

Have a great day! ✌️

**Comment by rhashimoto** [comment]
Thanks for the offer!

I tried TypeScript for a while but I never liked that [build step](https://xkcd.com/303/), so when ES6 module support reached a [dev server](https://modern-web.dev/guides/) and Chrome, I switched to plain Javascript. I know that loading modules is effectively a build but it was so much faster and simpler.

Nowadays my machine is a lot faster and I'm guessing tsc might be, too. And sometimes I wish that I could play with [decorators](https://github.com/tc39/proposal-decorators). And I do realize that some wa-sqlite forks exist only to fix some TypeScript mismatches. So although I'm quite happy writing in plain JavaScript, I want to give your proposal due consideration. Let me ponder this and maybe play around with workflows for a bit.

**Reply by schickling** [reply]
Thanks for sharing your thoughts and excited to hear you might be open to this. I have quite a lot of experience with TypeScript setups, so happy to help you think through some scenarios. Feel free to hit me up on Discord via `schickling`.
